### PR TITLE
React integration tests

### PR DESCRIPTION
**Pull request by davisshaver**
Adds a basic integration test confirming app launch.

Can be used in conjunction with server locally. I did so by `npm i -g npm-run-all` and then `run-p start test-integration -r --silent` so that the scripts run parallel and then stop.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -63,7 +66,7 @@
     "react-addons-test-utils": "15.4.1",
     "react-hot-loader": "3.0.0-beta.6",
     "react-test-renderer": "15.4.1",
-    "react-addons-test-utils": "15.4.1",
+    "selenium-download": "^2.0.7",
```
Comment (davisshaver) [inline package.json]: Dunno why this is diffed, `react-addons-test-utils` hasn't changed.

File: nightwatch.conf.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+const SCREENSHOT_PATH = "./tests/artifacts/screenshots/";
```
Comment (davisshaver) [inline nightwatch.conf.js]: Based on sample conf here: https://github.com/dwyl/learn-nightwatch

File: package.json [hunk 3]
```diff
@@ -10,7 +10,9 @@
     "locales": "node locales.js",
     "mock-api": "MOCKAPI=true DEVELOPMENT=true ./node_modules/babel-cli/bin/babel-node.js server.js",
     "test": "jest",
-    "test-watch": "jest --watch"
+    "test-watch": "jest --watch",
+    "test-integration": "nightwatch",
+    "postinstall": "node nightwatch.conf.js"
```
Comment (davisshaver) [inline package.json]: Ensures that Selenium is present

**Comment by joshkadis** [comment]
I'm getting 
```
Joshs-MacBook-Pro:simplechart joshkadis$ run-p start test-integration -r --silent
Starting selenium server... started - PID:  7401

[Simple] Test Suite
=======================

Running:  Simplechart assert title
 ✔ Element <body> was visible after 89 milliseconds.
 ✖ Testing if the page title equals "Simplechart".  - expected "Simplechart" but got: "localhost"
    at Object.module.exports.Simplechart assert title (/Users/joshkadis/Repos/simplechart/tests/integration/simple.js:8:12)


FAILED:  1 assertions failed and 1 passed (2.197s)

 _________________________________________________

 TEST FAILURE:  1 assertions failed, 1 passed. (2.284s)

 ✖ simple

   - Simplechart assert title (2.197s)
   Testing if the page title equals "Simplechart".  - expected "Simplechart" but got: "localhost"
       at Object.module.exports.Simplechart assert title (/Users/joshkadis/Repos/simplechart/tests/integration/simple.js:8:12)
   SKIPPED:
   - Simplechart assert basic use

Listening at http://localhost:8080/
```

Are the tests running before the local server is finished starting up?

**Comment by davisshaver** [comment]
@joshkadis Yes, looks like it. I got this once or twice, but it would usually work when I tried again. Lemme know if subsequent attempts work, if not we can try another approach. You can also run `npm run start` and `npm run test-integration` in separate terminal windows.

File: nightwatch.conf.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  if (err || !stat || stat.size < 1) {
+    require('selenium-download').ensure(BINPATH, function(error) {
+      if (error) throw new Error(error)
+      console.log('✔ Selenium & Chromedriver downloaded to:', BINPATH);
+    });
+  }
+});
+
+
+function padLeft (count) {
```
Comment (joshkadis) [inline nightwatch.conf.js]: Are you _sure_ there isn't a NPM package with ~30 dependencies that you could use instead?

**Comment by joshkadis** [comment]
Yeah, retries were successful. `npm run start & npm run test-integration` also works.

File: nightwatch.conf.js (hunk 4 above)
Comment (davisshaver) [inline nightwatch.conf.js]: 🐢

**Pull request merged**